tests/switchless_threads/host/host.c

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/tests.h>
+#include <openenclave/internal/thread.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#if __GNUC__
+#include <pthread.h>
+#elif _MSC_VER
+#include <windows.h>
+#endif
+#include "switchless_threads_u.h"
+
+// The enclave supports up to 8 concurrent threads in it. We have to reserve
```

Comment:
This statement isn't entirely true.  The number 8 in enc/enc.c is inside
OE_SET_ENCLAVE_SGX() which means it's only true for SGX.  The code here has no such limitation,
and we want switchless to apply to OP-TEE as well.  So minimally the comment should be fixed,
but ideally (not required for this PR), the code should be such that it won't accidentally compile for OP-TEE.
That is, compile time failures are better than runtime failures, until OP-TEE support is added.

Reply:
Fixed the comments.